src/Nethermind/Nethermind.AuRa/Validators/MultiValidator.cs

```diff
@@ -100,7 +102,7 @@ public void PreProcess(Block block, ProcessingOptions options = ProcessingOption
             {
                 if (TryGetLastValidator(previousBlockNumber, out var validatorInfo))
                 {
-                    if (validatorInfo.Value.ValidatorType.CanChangeImmediately())
+                    if (validatorInfo.Value.ValidatorType.CanChangeImmediately() || _blockFinalizationManager.LastFinalizedBlockLevel >= validatorInfo.Key)
```

Comment:
can we add some comment / explainer here

Reply:
Will name it in variable